pkg/paho-mqtt/contrib/riot_iface.c

```diff
@@ -0,0 +1,234 @@
+/*
+ * Copyright (C) 2019  Javier FILEIV <javier.fileiv@gmail.com>
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @brief       MQTT common RIOT interface functions
+ *
+ * @author      Javier FILEIV <javier.fileiv@gmail.com>
+ */
+#include <string.h>
+#include <errno.h>
+
+#ifdef MODULE_IPV6_ADDR
+#include <net/ipv6/addr.h>
+#endif
+
+#ifdef MODULE_IPV4_ADDR
+#include <net/ipv4/addr.h>
+#endif
+#include <net/af.h>
+#include <net/sock.h>
+#include <net/sock/tcp.h>
+#include <mqtt.h>
+#include "MQTTClient.h"
+#include <xtimer.h>
+#include <tsrb.h>
+#include <log.h>
```

Comment:
In accordance with our [coding conventions](https://github.com/RIOT-OS/RIOT/blob/master/CODING_CONVENTIONS.md#includes) please use quoted includes `"` for non-system headers and remove unneeded includes here and elsewhere (e.g. your `mqtt.h`):

```suggestion
#ifdef MODULE_IPV6_ADDR
#include "net/ipv6/addr.h"
#endif
#ifdef MODULE_IPV4_ADDR
#include "net/ipv4/addr.h"
#endif
#include "net/sock/tcp.h"
#include "mqtt.h"
#include "MQTTClient.h"
#include "xtimer.h"
#include "tsrb.h"
#include "log.h"
```